src/Service/Controllers/RestController.cs

```diff
@@ -224,16 +224,21 @@ public static JsonResult ErrorResponse(string code, string message, HttpStatusCo
 
                 if (result is CreatedResult)
                 {
-                    // Location is made up of two parts, the first being constructed
-                    // from the HttpRequest found in the HttpContext. The other part
-                    // is the primary key route, which has already been saved in the
+                    // Location is made up of three parts, the first being constructed
+                    // from the Host property found in the HttpContext.Request. The second part being the
+                    // base route configured in the config file.
+                    // The third part is the primary key route, which has already been saved in the
                     // Location of the created result. So we form the entire location
-                    // from appending the primary key route  already stored in the
+                    // from appending the base route and the primary key route  already stored in the
                     // created result to the url constructed from the HttpRequest. We
                     // then update the Location of the created result to this value.
                     CreatedResult createdResult = (result as CreatedResult)!;
-                    string location = UriHelper.GetEncodedUrl(HttpContext.Request) + "/" + createdResult.Location;
-                    createdResult.Location = location;
+                    string baseURL = UriHelper.BuildAbsolute(
```

Comment:
naming this 'baseURL' seems odd since we have a value called "BaseRoute" and 'base' implies that there is more URL to follow. Would 'locationUrl' make more sense? or something like that?

Reply:
Renamed